sklearn/neighbors/_base.py

```diff
@@ -614,9 +615,9 @@ class from an array representing our data set and ask who's
         n_samples_fit = self.n_samples_fit_
         if n_neighbors > n_samples_fit:
             raise ValueError(
-                "Expected n_neighbors <= n_samples, "
+                "Expected n_neighbors < n_samples, "
```

Comment:
I believe this new error message no longer reflects the original `X is not None` case.

|                  | `X is None`                          | `X is not None`                       |
|------------------|--------------------------------------|---------------------------------------|
| **Valid config** | `"Expected n_neighbors < n_samples"` | `"Expected n_neighbors <= n_samples"` |

Because there are two valid configs, I think that's why [the suggestion in the original issue](https://github.com/scikit-learn/scikit-learn/issues/17207#issuecomment-628692504) was to have 2 separate error messages depending on whether X was explicitly provided.

Reply:
Updated